### PR TITLE
Add typings for Tracing

### PR DESCRIPTION
**Pull request by vananabu**
## Related issue number
[#1749](https://github.com/aio-libs/aiohttp/issues/1749)
[](url)

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: aiohttp/tracing.py [hunk 1]
```diff
@@ -24,31 +30,37 @@ class TraceConfig:
...
-        self._on_dns_cache_miss = Signal(self)
-
-        self._trace_config_ctx_factory = trace_config_ctx_factory
-
-    def trace_config_ctx(self, trace_request_ctx=None):
+    def __init__(
+        self,
+        trace_config_ctx_factory: Type[SimpleNamespace]=SimpleNamespace
+    ) -> None:
+        self._on_request_start = Signal(self)  # type: Signal[Callable[['Application'], Awaitable[None]]] # noqa
```
Comment (kxepal) [inline aiohttp/tracing.py]: Maybe define `Signal[Callable[['Application'], Awaitable[None]]]` only once at top level and reuse it everywhere? You wouldn't need typing comments and qa disable in this case.

Comment (dbohomiakov) [reply aiohttp/tracing.py]: Refactored

File: aiohttp/tracing.py [hunk 2]
```diff
@@ -8,6 +9,15 @@
 from .signals import Signal
 
 
+if TYPE_CHECKING:
+    from .client import ClientSession
+    from .web_app import Application
+
+    _Signal = Signal[Callable[[Application], Awaitable[None]]]
```
Comment (asvetlov) [inline aiohttp/tracing.py]: Please replace `Application` with `TraceConfig` to fix the signature

Comment (dbohomiakov) [reply aiohttp/tracing.py]: done

File: aiohttp/tracing.py [hunk 3]
```diff
@@ -24,31 +33,37 @@ class TraceConfig:
...
-        self._on_dns_cache_miss = Signal(self)
-
-        self._trace_config_ctx_factory = trace_config_ctx_factory
-
-    def trace_config_ctx(self, trace_request_ctx=None):
+    def __init__(
+        self,
+        trace_config_ctx_factory: Type[SimpleNamespace]=SimpleNamespace
+    ) -> None:
+        self._on_request_start = Signal(self)  # type: _Signal
```
Comment (kxepal) [inline aiohttp/tracing.py:40]: Hm. Do we have any conventions about when to use type comment and when - annotation? Why not to use annotations everywhere? IIRC, those comments were made just for Python 2.x  typing support.

Comment (dbohomiakov) [reply aiohttp/tracing.py:40]: I didn't find any documented convention regarding style for annotations so it looks like the question to @asvetlov.

Comment (asvetlov) [reply aiohttp/tracing.py:40]: 1. This comment is for Python 3.5; `a: type = val` is supported by Python 3.6 only.
2. We are in the middle of the transition to a fully annotated code. After the work is done mypy strict mode will be enabled.

The line is correct.

Comment (kxepal) [reply aiohttp/tracing.py:40]: I almost forgot what 3.5 is. Thanks!

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**